activerecord/lib/active_record/relation/finder_methods.rb

```diff
@@ -459,7 +459,9 @@ def find_some(ids)
       end
 
       if result.size == expected_size
-        result
+        return result if self.values[:order]
+        records_by_id = result.index_by(&:id)
+        ids.first(expected_size).collect { |id| records_by_id[id.to_i] }
```

Comment:
Good call @jmbejar, it makes sense to me that it should work fine using `.limit` too, I'll work further on that.

Thanks for the heads-up and the test! :)